### DISCUSSION TITLE
移动网络下连接zerotier之后会丢失ipv6

### DISCUSSION THREAD

**Discussion by arcticblueoat** [kaaass/ZerotierFix#85] [Q&A] [answered]
在移动网络下链接zerotier之后 无法访问ipv6
在移动网络状态下绝大部分是用行星节点中继  只有几次是使用了ipv6  在不连接时是可以ping通另一个节点的ipv6的
  小米10s  最新稳定版miui
非专业人士  如果描述不准确还望海涵

**Comment by kaaass** [comment]
感谢报告，不过我暂时没能复现出类似的问题。

可否描述下已经做的测试和结果？比如 ping 了某设备。

**Comment by arcticblueoat** [comment]
没开启vpn时  我手机可以直接ping通装了zerotier的openwrt路由器的ipv6地址   手机也可以通过ipv6测试

这时候手机是有ipv6的
开启vpn之后   手机ping不通路由器的ipv6   也通不过ipv6 测试  显示我没有ipv6   同时zerotier也走的中继  
![Screenshot_2023-12-17-18-46-33-033_com android phone](https://github.com/kaaass/ZerotierFix/assets/108968067/21cacc1c-8d52-4cff-877c-63190b73658b)

这个时候就只有一个ipv4   还是我zerotier的内网网段
但是极少数时候是走的ipv6打洞的  

感谢回复

**Comment by arcticblueoat** [comment]
![IMG_20231217_184824](https://github.com/kaaass/ZerotierFix/assets/108968067/24f4a421-f60d-4832-b5e0-d24c132ae0cf)
这个是未连接vpn时的照片

**Comment by kaaass** [comment]
看描述感觉和 1.0.7 版本的 v6 问题很像（#51），可否检查下是否使用最新版本？

**Comment by arcticblueoat** [comment]
您好  我用的一直是最新版  几乎每周都来这看看是否更新

**Comment by kaaass** [comment]
#81 修复了一个可能和这个 Issue 有关的问题，可以试试那个 Issue 里面的版本是否还有这个问题 @arcticblueoat

**Comment by arcticblueoat** [comment]
您好 目前的问题是手机连接zt之后ipv6完全不可用  不论wifi还是数据流量  感觉是安卓（或者miui)系统的问题

**Comment by kaaass** [comment]
@arcticblueoat 试试这个包呢？[app-debug.apk.zip](https://github.com/kaaass/ZerotierFix/files/14205907/app-debug.apk.zip)

**Comment by Aqaao** [comment]
> @arcticblueoat 试试这个包呢？[app-debug.apk.zip](https://github.com/kaaass/ZerotierFix/files/14205907/app-debug.apk.zip)

也不行，自建planet，服务器版本 1.10.3。开了zerotierfix就ipv6测试连不通了，ping6 www.baidu.com 不通 no route to host。

**Comment by arcticblueoat** [comment]
您好 这两个包都用了很长时间   只有一次出现了ipv6直连  我推测可能是vpn打开后 手机的ip就变成了zt的内网ip  但是不知道为什么v6并没有成功分配  所以整个手机就无法使用v6了  以上是我的推测  希望帮到您

**Comment by kaaass** [comment]
我目前仍然没能复现，考虑到其他 Issue 中是有成功通过 Zt Fix 使用 v6 的（比如 RDP 远程桌面），可能触发问题需要一定前置条件。方便的话可以列举一下完整的步骤，例如：

1. `ping6 xxxxx` 成功
2. 打开 Zt Fix APP，开启网络，等待通知栏出现 “成功连接”
3. `ping6 xxxxx` 失败，显示 `xxxx`

**Comment by arcticblueoat** [comment]
您看看这些够不够
![Screenshot_2024-04-05-00-13-27-036_com ddm iptools](https://github.com/kaaass/ZerotierFix/assets/108968067/4d844555-1429-412b-aac9-85eba86a67ae)
![Screenshot_2024-04-05-00-13-44-098_net kaaass zerotierfix](https://github.com/kaaass/ZerotierFix/assets/108968067/4bb7b2ec-0d71-49ab-b582-26583f28d2af)
![Screenshot_2024-04-05-00-14-48-689_com ddm iptools](https://github.com/kaaass/ZerotierFix/assets/108968067/d957c0bc-0bf7-4b6d-b651-7d51ef1f7df3)

**Comment by kaaass** [comment]
点击 APP 中的网络看下详细信息，能否在 “分配 IP” 的区域中看到 IPv6 的地址呢？（如果要截图记得打码）

**Comment by arcticblueoat** [comment, marked as answer]
我把控制器的v6分配关闭了 重新打开   就能分配到v6     v6的网址也能正常访问了   非常感谢   求教一下  正常那三个v6选项一般要用哪个呢？

**Reply by kaaass** [reply]
三个差异都不是很大，主要的影响是地址的构成格式。我自己一般用的是 6PLANE，这个模式下可以拿到 /80 的地址，地址比较短而且可以作为前缀继续分配。此外这几个模式应该也是可以多选的，那样会分配多个 v6 地址。

**Comment by kaaass** [comment]
没事的～ 因为不是 Issue 我就先转 Discussion 了，不影响后续讨论
